Query/EncloseNames.php

```diff
@@ -0,0 +1,141 @@
+ *     * Neither the name of the Hoa nor the names of its contributors may be
+ *       used to endorse or promote products derived from this software without
+ *       specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDERS AND CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ */
+
+namespace Hoa\Database\Query;
+
+/**
+ * Abstract class \Hoa\Database\Query\EncloseNames.
+ *
+ * Enclose names feature.
+ *
+ * @copyright  Copyright © 2007-2015 Hoa community
+ * @license    New BSD License
+ */
+abstract class EncloseNames
+{
+    /**
+     * State of the enclose names feature.
+     *
+     * @var bool
+     */
+    protected $_enableEnclose = false;
+
+    /**
+     * Enclose opening symbol.
+     *
+     * @var string
+     */
+    protected $_openingSymbol = '"';
+
+    /**
+     * Enclose closing symbol.
+     *
+     * @var string
+     */
+    protected $_closingSymbol = '"';
+
+
+
+    /**
+     * Set enclose symbols.
+     *
+     * @param  string  $openingSymbol    Opening symbol.
+     * @param  string  $closingSymbol    Closing symbol.
+     * @return  EncloseNames
+     */
+    public function setEncloseSymbol($openingSymbol, $closingSymbol = null)
+    {
+        $this->_openingSymbol = $openingSymbol;
+        $this->_closingSymbol =
+            (null === $closingSymbol)
+                ? $openingSymbol
+                : $closingSymbol;
+
+        return $this;
+    }
+
+    /**
+     * Enable or disable enclose names feature.
+     *
+     * @param  bool  $enable    Enable or disable.
+     * @return  bool
+     */
+    public function enableEncloseNames($enable = true)
+    {
+        $old                  = $this->_enableEnclose;
+        $this->_enableEnclose = $enable;
+
+        return $old;
+    }
+
+    /**
+     * Enclose table/column/alias names with defined symbol.
+     *
+     * @param  array|string  $names    Table/column/alias names.
+     * @return  array|string
+     */
+    protected function enclose($names)
+    {
+        if (false === $this->_enableEnclose) {
+            return $names;
+        }
+
+        if (false === is_array($names)) {
+            return $this->_enclose($names);
+        }
+
+        foreach ($names as &$name) {
```

Comment:
the return will be an array and a string is expected if we pass a string in input.